File: csharp/main/UInt16Def.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class UInt16Def : IConstantScope, ITypeDef<TypeUInt16>
	{
		private UInt16Def()
		{
			DefinedConstants.Add(new ConstantDef(this, "MIN_VALUE", TypeUInt16.Instance, new LiteralUInt16(UInt16.MinValue)));
			DefinedConstants.Add(new ConstantDef(this, "MAX_VALUE", TypeUInt16.Instance, new LiteralUInt16(UInt16.MaxValue)));
		}

		public static UInt16Def Instance = new UInt16Def();

		public String Name => "UInt16";

		public String FullName => Name;

		public List<ConstantDef> DefinedConstants { get; } = new List<ConstantDef>();

		public TypeUInt16 Type => TypeUInt16.Instance;

		public FileDef File => null;

		public ITypeScope Parent => null;
	}
}

File: java/src/main/java/deltix/luminary/UInt16Def.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public class UInt16Def implements TypeDef<TypeUInt16>, ConstantScope {
    private List<ConstantDef> definedConstants;

    private UInt16Def() {
        definedConstants = new ArrayList<>();
        definedConstants.add(new ConstantDef(this, "MIN_VALUE", TypeUInt16.INSTANCE, new LiteralUInt16((short) 0x00_00)));
        definedConstants.add(new ConstantDef(this, "MAX_VALUE", TypeUInt16.INSTANCE, new LiteralUInt16((short) 0xFF_FF)));
    }

    public static final UInt16Def INSTANCE = new UInt16Def();

    @Override
    public String getName() {
        return "UInt16";
    }

    @Override
    public String getFullName() {
        return getName();
    }

    @Override
    public List<ConstantDef> getDefinedConstants() {
        return definedConstants;
    }

    @Override
    public TypeUInt16 getType() {
        return TypeUInt16.INSTANCE;
    }

    @Override
    public FileDef getFile() {
        return null;
    }

    @Override
    public TypeScope getParent() {
        return null;
    }
}
